sovereign-forge/tests/sovereign_forge_unit_tests.rs

```diff
@@ -354,6 +418,192 @@ fn register_chain_factory() {
         });
 }
 
+#[test]
+fn update_sovereign_config_no_chain_config_deployed() {
+    let mut state = SovereignForgeTestState::new();
+    state.deploy_sovereign_forge();
+
+    state.register_chain_factory(2, TOKEN_HANDLER_ADDRESS, None);
+
+    state
+        .world
+        .query()
+        .to(FORGE_ADDRESS)
+        .whitebox(sovereign_forge::contract_obj, |sc| {
+            assert!(!sc.chain_factories(2).is_empty());
+        });
+
+    state.update_sovereign_config(
+        SovereignConfig::default_config(),
+        Some("The current caller has not deployed any Sovereign Chain"),
+    );
+}
+
+#[test]
+fn update_sovereign_config() {
+    let mut state = SovereignForgeTestState::new();
+    state.deploy_sovereign_forge();
+    state.deploy_chain_factory();
+    state.deploy_chain_config_template();
+    state.deploy_fee_market_template();
+    state.finish_setup();
+
+    state
+        .world
+        .query()
+        .to(FORGE_ADDRESS)
+        .whitebox(sovereign_forge::contract_obj, |sc| {
+            assert!(!sc.chain_factories(2).is_empty());
+        });
+
+    let deploy_cost = BigUint::from(100_000u32);
+
+    state.complete_setup_phase(None);
+
+    state.deploy_phase_one(
+        &deploy_cost,
+        Some(ManagedBuffer::from("SVCH")),
```

Comment:
This is prefered_chain_id but it needs to be 1-4 alphanumeric lowercase characters. We need to check this in `generate_chain_id` function